rotkehlchen/exchanges/coinbase.py

```diff
@@ -285,34 +382,40 @@ def _api_query(
             options: dict[str, Any] | None = None,
             ignore_pagination: bool = False,
     ) -> list[Any]:
-        """Performs a coinbase API Query for endpoint
-
-        You can optionally provide extra arguments to the endpoint via the options argument.
-        If you want just the first results then set ignore_pagination to True.
-        """
         all_items: list[Any] = []
         had_4xx = False
         request_verb = 'GET'
-        # initialize next_uri before loop
         next_uri = f'/{self.apiversion}/{endpoint}'
         timeout = CachedSettings().get_timeout_tuple()
+        
+
         if options:
             next_uri += f'?{urlencode(options)}'
         while True:
-            timestamp = str(int(time.time()))
-            message = timestamp + request_verb + next_uri
-
-            signature = hmac.new(
-                self.secret,
-                message.encode(),
-                hashlib.sha256,
-            ).hexdigest()
-            log.debug('Coinbase API query', request_url=next_uri)
-
-            self.session.headers.update({
-                'CB-ACCESS-SIGN': signature,
-                'CB-ACCESS-TIMESTAMP': timestamp,
-            })
+            if self.is_legacy_api_key:
+                timestamp = str(int(time.time()))
+                message = timestamp + request_verb + next_uri
+
+                signature = hmac.new(
+                    self.secret,
+                    message.encode(),
+                    hashlib.sha256,
+                ).hexdigest()
+                log.debug('Coinbase API query', request_url=next_uri)
+
+                self.session.headers.update({
+                    'CB-ACCESS-SIGN': signature,
+                    'CB-ACCESS-TIMESTAMP': timestamp,
+                })
+            else:
+                request_path = f'{self.host}/{self.apiversion}/{endpoint}'
+                log.debug('CDP Request Path ', request_url=request_path)
+                uri = f'{request_verb} {request_path}'
+                log.debug('CDP URI Path ', request_url=uri)
```

Comment:
Don't spam the logs with multiple log entries for the samething.

```suggestion
                uri = f'{request_verb} {self.host}/{self.apiversion}/{endpoint}'
                log.debug('Coinbase CDP API query ', request_url=uri)
```